docs/src/contributing/development-environment.md

```diff
@@ -1,280 +1,278 @@
 
-Additionally, you may need:
-
-- [shellcheck](https://github.com/koalaman/shellcheck)
-- [golint](https://github.com/golang/lint)
+- Python 2.7 or later (with `pip`)
+- virtualenv (`sudo pip install virtualenv`)
 
 In most cases, you should simply install according to the instructions. There are a few special cases, though. We cover these below.
 
 ### Configuring Go
 
-If your local workstation does not support the `linux/amd64` target environment, you will have to install Go from source with cross-compile support for that environment. This is because some of the components are built on your local machine and then injected into a docker container.
+If your local workstation does not support the `linux/amd64` target environment, you will have to install Go from source with cross-compile support for that environment. This is because some of the components are built on your local machine and then injected into a Docker container.
 
 Homebrew users can just install with cross compiling support:
 
-
-    $ brew install go --with-cc-common
+```
+$ brew install go --with-cc-common
+```
 
 It is also straightforward to build Go from source:
 
-    $ sudo su
-    $ curl -sSL https://golang.org/dl/go1.5.src.tar.gz | tar -v -C /usr/local -xz
-    $ cd /usr/local/go/src
-    $ # compile Go for our default platform first, then add cross-compile support
-    $ ./make.bash --no-clean
-    $ GOOS=linux GOARCH=amd64 ./make.bash --no-clean
+```
+$ sudo su
+$ curl -sSL https://golang.org/dl/go1.5.src.tar.gz | tar -v -C /usr/local -xz
+$ cd /usr/local/go/src
+$ # compile Go for our default platform first, then add cross-compile support
+$ ./make.bash --no-clean
+$ GOOS=linux GOARCH=amd64 ./make.bash --no-clean
+```
 
-Once you can compile to `linux/amd64`, you should be able to compile Deis' components as normal.
+Once you can compile to `linux/amd64`, you should be able to compile Deis components as normal.
 
 ### Configuring Docker Machine (Mac)
 
-Deis needs a Docker registry running independently of the Deis cluster. On OS X, you will need [Docker Machine][machine] to run the registry inside of a VirtualBox image.
+Deis needs Docker for building images.  Docker utilizes a client/server architecture, and while the Docker client is available for Mac OS, the Docker server is dependent upon the Linux kernel.  Therefore, in order to use Docker on Mac OS, [Docker Machine][machine] is used to facilitate running the Docker server within a [VirtualBox][vbox] VM.
 
-!!! note
-    Previously, Deis used boot2docker to run the registry. However, Docker has deprecated boot2docker in favor of Docker Machine.
+Install Docker Machine according to the normal installation instructions, then use it to create a new VM:
 
-Install Docker Machine according to the normal installation instructions. Then create a new image for hosting your Deis Docker registry:
+```
+$ docker-machine create deis-docker \
+    --driver=virtualbox \
```

Comment:
Are we assuming that users have VirtualBox installed? I think it was removed as a dependency above.